database/migrations/default/1729616317798_set_more_spatial_attributes/up.sql

```diff
@@ -0,0 +1,134 @@
+
+CREATE OR REPLACE FUNCTION public.crashes_set_spatial_attributes()
+ RETURNS trigger
+ LANGUAGE plpgsql
+AS $function$
+begin
+    if (new.latitude is not null and new.longitude is not null) then
+        -- save lat/lon into geometry col
+        new.position = st_setsrid(st_makepoint(new.longitude, new.latitude), 4326);
+        --
+        -- get location polygon id
+        --
+        if (new.rpt_road_part_id != 2 and upper(ltrim(new.rpt_hwy_num)) in ('35', '183','183A','1','290','71','360','620','45','130')) then
+            -- use level 5 polygon
+            new.location_id = (
+                select
+                    location_id
+                from
+                    public.atd_txdot_locations
+                where
+                    location_group = 2 -- level 5
+                    and st_contains(geometry, new.position)
+                limit 1);
+        else
+            -- use the other polygons
+            new.location_id = (
+                select
+                    location_id
+                from
+                    public.atd_txdot_locations
+                where
+                    location_group = 1 -- not level 5
+                    and st_contains(geometry, new.position)
+                limit 1);
+        end if;
+
+        raise debug 'found location: % compared to previous location: %', new.location_id, old.location_id;
+        --
+        -- check if in austin full purpose jurisdiction
+        --
+        new.in_austin_full_purpose =  st_contains((select geometry from geo.atd_jurisdictions where id = 5), new.position);
+        raise debug 'in austin full purpose: % compared to previous: %', new.in_austin_full_purpose, old.in_austin_full_purpose;
+        --
+        -- get council district
+        --
+        new.council_district = (
+            select
+                council_district
+            from
+                geo.council_districts
+            where
+                st_contains(geometry, new.position)
+            limit 1);
+        raise debug 'council_district: % compared to previous: %', new.council_district, old.council_district;
+        --
+        -- get engineering area
+        --
+        new.engineering_area_id = (
+            select
+                engineering_area_id
+            from
+                geo.engineering_areas
+            where
+                st_contains(geometry, new.position)
+            limit 1);
+        raise debug 'engineering_area_id: % compared to previous: %', new.engineering_area_id, old.engineering_area_id;
+        --
+        -- get signal engineer area
+        --
+        new.signal_engineer_area_id = (
+            select
+                signal_engineer_area_id
+            from
+                geo.signal_engineer_areas
+            where
+                st_contains(geometry, new.position)
```

Comment:
i was going to suggest to make this more efficient you could do 
`where (geometry && new.position and (st_contains(geometry, new.position))`
because ive been given that piece of advice in the past, but i just did a little digging and it seems like `st_contains` [automatically does a bounding box](https://postgis.net/docs/ST_Contains.html#:~:text=This%20function%20automatically%20includes%20a%20bounding%20box%20comparison%20that%20makes%20use%20of%20any%20spatial%20indexes%20that%20are%20available%20on%20the%20geometries.%20To%20avoid%20index%20use%2C%20use%20the%20function%20_ST_Contains.) `&&` check before moving on to the other part of the function, so adding an `&&` before calling `st_contains` would be [redundant](https://dba.stackexchange.com/questions/191666/vs-st-intersects-performance#:~:text=The%20reason%20why%20is%20simple%2C%20only%20the%20bounding%20box%20uses%20the%20index.%20That%20means%20it%27ll%20do%20a%20%26%26%20AND%20something%20else.%20And%20you%20can%20see%20that%20with%20%5CdfS%2B%20st_intersects). now im thinking of all the places in our code where we do this redundant double bounding box check. super curious if im understanding this correctly after doing my little rabbit hole dive

Reply:
yes, that's right—if you're already using spatial indexes than that's not necessary. i don't think there are too many places where `&&` is still around except perhaps in the old location triggers. we will clean them up eventually...